lunavl/sdk/faceengine/facedetector.py

```diff
@@ -372,41 +365,34 @@ def redetectOne(self, image, *, bBox=None, detection=None, detect5Landmarks=True
             detect68Landmarks: detect or not landmarks68
 
         Returns:
-            detection
+            detection if face found otherwise None
         Raises:
             LunaSDKException if an error occurs
         """
-        if isinstance(image, ImageForRedetection) and (bBox is detection is None):
-            # todo remove \\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\ on 3.9.0 release
-            imageForRedetection = ImageForRedetection(image.image, DetectionFloat(image.bBox.coreRectF, 1.0))
-            # todo remove ///////////////////////////////////////
-            error, detectRes = self._detector.redetectOne(imageForRedetection.image.coreImage, imageForRedetection.bBox,
-                                                          self._getDetectionType(detect5Landmarks, detect68Landmarks))
-            # error, detectRes = self._detector.redetectOne(image.image.coreImage, image.BBox.coreRectF,
-            #                                               self._getDetectionType(detect5Landmarks, detect68Landmarks))
-            vlImage = image.image
-        elif isinstance(image, VLImage) and (bBox is not None) and (detection is None):
+        if isinstance(image, VLImage) and (bBox is not None) and (detection is None):
             error, detectRes = self._detector.redetectOne(image.coreImage, bBox.coreRectF,
                                                           self._getDetectionType(detect5Landmarks, detect68Landmarks))
             vlImage = image
         elif isinstance(image, VLImage) and (bBox is None) and (detection is not None):
-            error, detectRes = self._detector.redetectOne(image.coreImage, detection,
+            error, detectRes = self._detector.redetectOne(image.coreImage, detection.coreEstimation.detection.rect,
                                                           self._getDetectionType(detect5Landmarks, detect68Landmarks))
             vlImage = image
         else:
             raise NotImplementedError
 
         if error.isError:
             raise LunaSDKException(LunaVLError.fromSDKError(error))
-        return FaceDetection(detectRes, vlImage)
+        if detectRes.isValid():
+            return FaceDetection(detectRes, vlImage)
+        return None
 
     @CoreExceptionWrap(LunaVLError.DetectFacesError)
     def redetect(
             self,
             images: List[ImageForRedetection],
             detect5Landmarks: bool = True,
             detect68Landmarks: bool = False,
-    ) -> List[FaceDetection]:
+    ) -> List[List[Union[FaceDetection, None]]]:
```

Comment:
```suggestion
    ) -> List[List[Optional[FaceDetection]]]:
```
https://docs.python.org/3/library/typing.html#typing.Optional